src/request/sign-transaction/SignTransaction.js

```diff
@@ -46,19 +49,37 @@ class SignTransaction {
 
         /** @type {HTMLLinkElement} */
         const $recipient = (this.$el.querySelector('.accounts .recipient'));
+        const recipientAddress = transaction.recipient.toUserFriendlyAddress();
+        const recipientLabel = 'shopOrigin' in request && !!request.shopOrigin
+            ? request.shopOrigin.split('://')[1]
+            : request.recipientLabel || null;
         this._recipientAddressInfo = new AddressInfo({
-            userFriendlyAddress: transaction.recipient.toUserFriendlyAddress(),
-            label: request.shopOrigin
-                ? request.shopOrigin.split('://')[1]
-                : request.recipientLabel || null,
-            imageUrl: request.shopLogoUrl || null,
+            userFriendlyAddress: recipientAddress,
+            label: recipientLabel,
+            imageUrl: 'shopLogoUrl' in request && !!request.shopLogoUrl
+                ? request.shopLogoUrl
+                : null,
             accountLabel: null,
```

Comment:
I think for consistency the image assignment conditional should also be pulled above the object.